.pre-commit-config.yaml

```diff
@@ -66,6 +66,14 @@ repos:
     stages: [manual]
     pass_filenames: false
 
+  - id: check-unused-components
```

Comment:
Have you considered implementing the check as a bazel test rule? It could then be run with bazel like any test (`bazel test ...`) and would be automatically included in the CI pipeline. Bazel would run the test in parallel with other tests.